### PR TITLE
feat: add kusion server, engine api and refactor preview logic

### PR DESCRIPTION
**Pull request by ffforest**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ffforest** [comment]
Sample usage(requires workspace config to be present currently, should automatically query in future versions):
http://localhost:8888/api/v1/project/simple-service-4/stack/dev/preview?path=/Users/forest/playground/simple-service-4&isKCLPackage=true&detail=true&output=json
image
http://localhost:8888/api/v1/project/simple-service-4/stack/dev/preview?path=/Users/forest/playground/simple-service-4&isKCLPackage=true
image

File: air.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+root = "."
+tmp_dir = "tmp"
+
+[build]
+  # Just plain old shell command. You could use `make` as well.
+  cmd = "make build-local-darwin-arm64"
+  # Binary file yields from `cmd`.
+  bin = "./_build/bundles/kusion-darwin-arm64/bin/kusion"
+  # Customize binary, can setup environment variables when run your app.
+  # full_bin = "APP_ENV=dev APP_USER=air ./build/air/karma"
```
Comment (elliotxx) [inline air.toml:18]: need to remove `karma`

File: pkg/cmd/apply/options.go [hunk 2]
```diff
@@ -98,7 +99,9 @@ func (o *Options) Run() error {
 	}
 
 	// Compute changes for preview
-	changes, err := previewcmd.Preview(&o.Options, stateStorage, sp, project, stack)
+	// changes, err := previewcmd.Preview(&o.Options, stateStorage, sp, project, stack)
+	previewOptions := engineapi.NewAPIOptions()
```
Comment (elliotxx) [inline pkg/cmd/apply/options.go:103]: I see the `engineapi.NewAPIOptions() != &o.Options`

File: pkg/cmd/server/server.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+func NewCmdServer() *cobra.Command {
+	var (
+		serverShort = i18n.T(`start kusion server`)
+
+		serverLong = i18n.T(`Start kusion server.`)
+
+		serverExample = i18n.T(`
+		# Start kusion server
+		kusion server start --mode kcp`)
```
Comment (elliotxx) [inline pkg/cmd/server/server.go:19]: Too many spaces.

File: pkg/engine/api/builders/kcl/kcl_builder.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"path"
```
Comment (elliotxx) [inline pkg/engine/api/builders/kcl/kcl_builder.go:9]: Why is `path`, not `path/filepath`

File: pkg/engine/operation/models/change.go [hunk 5]
```diff
@@ -44,17 +44,71 @@ func (cs *ChangeStep) Diff() (string, error) {
...
+		return "", err
+	}
+
+	buf := bytes.NewBufferString("")
+
 	if len(cs.ID) != 0 {
-		buf.WriteString(pretty.GreenBold("ID: "))
-		buf.WriteString(pretty.Green("%s\n", cs.ID))
+		buf.WriteString("ID: ")
+		buf.WriteString(fmt.Sprintf("%s\n", cs.ID))
```
Comment (elliotxx) [inline pkg/engine/operation/models/change.go:103]: `s/fmt.Sprintf("%s\n", cs.ID)/cs.ID + "\n"`

File: pkg/server/route/route.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+// resource type and setting up proper handlers.
+func setupRestAPIV1(
+	r chi.Router,
+) {
+	r.Route("/project", func(r chi.Router) {
+		r.Route("/{projectName}", func(r chi.Router) {
+			r.Route("/stack", func(r chi.Router) {
+				r.Route("/{stackName}", func(r chi.Router) {
+					r.Route("/preview", func(r chi.Router) {
+						r.Get("/", stackhandler.ExecutePreview())
```
Comment (elliotxx) [inline pkg/server/route/route.go:63]: Soooo Deep Orz

File: air.toml (hunk 1 above)
Comment (ffforest) [reply air.toml:18]: Done

File: pkg/engine/api/builders/kcl/kcl_builder.go (hunk 4 above)
Comment (ffforest) [reply pkg/engine/api/builders/kcl/kcl_builder.go:9]: This is moved directly from the `cmd` pkg

File: pkg/engine/operation/models/change.go (hunk 5 above)
Comment (ffforest) [reply pkg/engine/operation/models/change.go:103]: Done

File: pkg/server/route/route.go (hunk 6 above)
Comment (ffforest) [reply pkg/server/route/route.go:63]: This is not final but I'm thinking this should be a stack-level operation, next to stack CRUD.
There should also be project level CRUD.

File: pkg/cmd/apply/options.go (hunk 2 above)
Comment (ffforest) [reply pkg/cmd/apply/options.go:103]: Right. `o.Options` is the command line options for the preview command, it's different from `engineapi.NewAPIOptions()` which creates a new engine api option (current it looks like a subset of the command line options).

**Pull request closed without merging**